jib-core/src/main/java/com/google/cloud/tools/jib/api/JavaContainerBuilder.java

```diff
@@ -465,23 +542,30 @@ public JibContainerBuilder toContainerBuilder() throws IOException {
             .filter(entry -> entry.getValue() > 1)
             .map(Entry::getKey)
             .collect(Collectors.toList());
-    for (Path file : addedDependencies) {
-      // Add dependencies to layer configuration
-      addFileToLayer(
-          layerBuilders,
-          file.getFileName().toString().contains("SNAPSHOT")
-              ? LayerType.SNAPSHOT_DEPENDENCIES
-              : LayerType.DEPENDENCIES,
-          file,
-          appRoot
-              .resolve(dependenciesDestination)
-              .resolve(
-                  duplicates.contains(file.getFileName().toString())
-                      ? file.getFileName()
-                              .toString()
-                              .replaceFirst("\\.jar$", "-" + Files.size(file))
-                          + ".jar"
-                      : file.getFileName().toString()));
+
+    ImmutableMap<LayerType, List<Path>> layerMap = ImmutableMap.of(
+        LayerType.DEPENDENCIES, addedDependencies,
+        LayerType.SNAPSHOT_DEPENDENCIES, addedSnapshotDependencies,
+        LayerType.PROJECT_DEPENDENCIES, addedProjectDependencies
+    );
+    for (LayerType layerType : layerMap.keySet()) {
+      for (Path file : Preconditions.checkNotNull(layerMap.get(layerType))) {
+        // Add dependencies to layer configuration
+        addFileToLayer(
+            layerBuilders,
+            layerType,
+            file,
+            appRoot
+                .resolve(dependenciesDestination)
+                // handle duplicates by appending filesize to the end of the file
+                .resolve(
+                    duplicates.contains(file.getFileName().toString())
```

Comment:
`file.getFileName().toString()` is repeating. Have a local variable?

Reply:
Huh, not sure why this is missing, but I remember I did

```java
      // Add dependencies to layer configuration
      String jarName = file.getFileName().toString();
      if (duplicates.contains(jarName)) {
        jarName = jarName.replaceFirst("\\.jar$", "-" + Files.size(file)) + ".jar";
      }
      addFileToLayer(
          layerBuilders,
          jarName.contains("SNAPSHOT") ? LayerType.SNAPSHOT_DEPENDENCIES : LayerType.DEPENDENCIES,
          file,
          appRoot.resolve(dependenciesDestination).resolve(jarName));
```

So I prefer the above.